server/src/main/java/org/opensearch/search/aggregations/MultiBucketConsumerService.java

```diff
@@ -131,12 +131,8 @@ public static class MultiBucketConsumer implements IntConsumer {
         private final int limit;
         private final CircuitBreaker breaker;
 
-        // aggregations execute in a single thread for both sequential
-        // and concurrent search, so no atomic here
-        private int count;
-
-        // will be updated by multiple threads in concurrent search
-        // hence making it as LongAdder
+        // count and callCount will both be updated by multiple threads in concurrent segment search hence using LongAdder
+        private final LongAdder count;
```

Comment:
not sure why we need to make `count` a LongAdder as this is only used in final reduce phase on coordinator. There accept is called with non-zero size value. Ref [here](https://github.com/opensearch-project/OpenSearch/issues/7785#issuecomment-1574337985)

Reply:
I reverted the `LongAdder` change in the latest revision as `count` is definitely not being used concurrently currently. I will look into the max_buckets limit some more and create a new issue for that if needed.

Reply:
Thought on this some more and I think it's fine actually. The purpose of the `search.max_buckets` setting seems to be to control the number of buckets returned globally to the whole request. For concurrent segment search we may have a larger number of intermediary buckets as we perform `reduce` on the slice level results, but it doesn't really make sense to enforce any limit in that phase.

Reply:
Thanks for checking this. I think we can still have this check of max bucket settings limit for slice level reduce. That way if any shard is exceeding the bucket count, we can fail it there and avoid overloading the coordinator. Have created a tracking issue for this improvement: https://github.com/opensearch-project/OpenSearch/issues/12916